worldedit-core/src/main/java/com/sk89q/worldedit/WorldEdit.java

```diff
@@ -396,8 +407,10 @@ private File getSafeFileWithExtension(File dir, String filename, String extensio
         return new File(dir, filename);
     }
 
+    private static final java.util.regex.Pattern SAFE_FILENAME_REGEX = java.util.regex.Pattern.compile("^[A-Za-z0-9_\\- \\./\\\\'\\$@~!%\\^\\*\\(\\)\\[\\]\\+\\{\\},\\?]+\\.[A-Za-z0-9]+$");
+
     private boolean checkFilename(String filename) {
-        return filename.matches("^[A-Za-z0-9_\\- \\./\\\\'\\$@~!%\\^\\*\\(\\)\\[\\]\\+\\{\\},\\?]+\\.[A-Za-z0-9]+$");
+        return SAFE_FILENAME_REGEX.matcher(filename).matches();
```

Comment:
This set of characters doesn't make sense, it's not safe for windows (contains `?`, `*`, doesn't ban COM and others), and POSIX will generally take anything. It also excludes extended Unicode characters for some reason, despite those being perfectly fine.

Can we just drop checking this aside from "you must have an extension", i.e. `^.+\..+$`?